torchvision/models/_utils.py

```diff
@@ -48,7 +48,7 @@ def __init__(self, model, return_layers):
         super(IntermediateLayerGetter, self).__init__()
 
         orig_return_layers = return_layers
-        return_layers = {k: v for k, v in return_layers.items()}
+        return_layers = {str(k): str(v) for k, v in return_layers.items()}
```

Comment:
This will need changing a few things downstream I think, so that the state_dicts are still compatible

Reply:
I don't know if this is the right change. This was an unsuccessful attempt to handle `return_layers = {'layer1': 0, 'layer2': 1, 'layer3': 2, 'layer4': 3}` but i wasn't able to change convert the values here to strings without breaking things. That still needs to be handled somehow...

Reply:
We would need to convert a ModuleList to a ModuleDict I think

Reply:
IntermediateLayerGetter is already a ModuleDict - what change were you thinking of ?

Reply:
Ok, I think I know what's going on.

We will need to change https://github.com/pytorch/vision/blob/f0d3daa7f65bcde560e242d9bccc284721368f02/torchvision/models/detection/faster_rcnn.py#L202
to use strings instead `['0', '1', '2', '3']`, and similarly in https://github.com/pytorch/vision/blob/f0d3daa7f65bcde560e242d9bccc284721368f02/torchvision/models/detection/mask_rcnn.py#L181 and https://github.com/pytorch/vision/blob/f0d3daa7f65bcde560e242d9bccc284721368f02/torchvision/models/detection/keypoint_rcnn.py#L182